tests/test_export.py

```diff
@@ -0,0 +1,256 @@
+        measurementspec['config']['parameters'][0]['sigmas'][0]
+    )
+    poi = m.find('POI')
+    assert poi is not None
+    assert poi.text == measurementspec['config']['poi']
+    paramsetting = m.find('ParamSetting')
+    assert paramsetting is None
+
+
+@pytest.mark.parametrize(
+    "spec, has_root_data, attrs",
+    [
+        (spec_staterror(), True, ['Activate', 'HistoName']),
+        (spec_histosys(), True, ['HistoNameHigh', 'HistoNameLow']),
+        (spec_normsys(), False, ['High', 'Low']),
+        (spec_shapesys(), True, ['ConstraintType', 'HistoName']),
+    ],
+    ids=['staterror', 'histosys', 'normsys', 'shapesys'],
+)
+def test_export_modifier(mocker, spec, has_root_data, attrs):
+    channelspec = spec['channels'][0]
+    channelname = channelspec['name']
+    samplespec = channelspec['samples'][1]
+    samplename = samplespec['name']
+    sampledata = samplespec['data']
+    modifierspec = samplespec['modifiers'][0]
+
+    mocker.patch('pyhf.writexml._ROOT_DATA_FILE')
+    modifier = pyhf.writexml.build_modifier(
+        modifierspec, channelname, samplename, sampledata
+    )
+    assert modifier.attrib['Name'] == modifierspec['name']
+    assert all(attr in modifier.attrib for attr in attrs)
+    assert pyhf.writexml._ROOT_DATA_FILE.__setitem__.called == has_root_data
+
+
+@pytest.mark.parametrize(
+    "spec",
+    [spec_staterror(), spec_histosys(), spec_normsys(), spec_shapesys()],
+    ids=['staterror', 'histosys', 'normsys', 'shapesys'],
+)
+def test_export_sample(mocker, spec):
+    channelspec = spec['channels'][0]
+    channelname = channelspec['name']
+    samplespec = channelspec['samples'][1]
+    samplename = samplespec['name']
+    sampledata = samplespec['data']
+
+    mocker.patch('pyhf.writexml.build_modifier', return_value=ET.Element("Modifier"))
+    mocker.patch('pyhf.writexml._ROOT_DATA_FILE')
+    sample = pyhf.writexml.build_sample(samplespec, channelname)
+    assert sample.attrib['Name'] == samplespec['name']
+    assert sample.attrib['HistoName']
+    assert sample.attrib['InputFile']
+    assert sample.attrib['NormalizeByTheory'] == str(False)
+    assert pyhf.writexml.build_modifier.called
+    assert pyhf.writexml._ROOT_DATA_FILE.__setitem__.called
+
+
+@pytest.mark.parametrize(
+    "spec",
+    [spec_staterror(), spec_histosys(), spec_normsys(), spec_shapesys()],
+    ids=['staterror', 'histosys', 'normsys', 'shapesys'],
+)
+def test_export_channel(mocker, spec):
+    channelspec = spec['channels'][0]
+    channelname = channelspec['name']
+
+    mocker.patch('pyhf.writexml.build_data', return_value=ET.Element("Data"))
+    mocker.patch('pyhf.writexml.build_sample', return_value=ET.Element("Sample"))
+    mocker.patch('pyhf.writexml._ROOT_DATA_FILE')
+    channel = pyhf.writexml.build_channel(channelspec, {})
+    assert channel.attrib['Name'] == channelspec['name']
+    assert channel.attrib['InputFile']
+    assert pyhf.writexml.build_data.called == False
```

Comment:
Can we use the [CodeFactor suggestion](https://www.codefactor.io/repository/github/diana-hep/pyhf/pull/426) here RE: "Comparison to False should be 'expr is False'"?